vm/simple_server_test.go

```diff
@@ -28,6 +31,53 @@ func TestServerInitialization(t *testing.T) {
 	}
 }
 
+func TestServerSetupResponse(t *testing.T) {
+	tests := []struct {
+		input    string
+		expectedBody string
+		expectedStatus int
+	}{
+		{`
+		require "net/simple_server"
+
+		server = Net::SimpleServer.new(4000)
+		server.get("/") do |req, res|
+		  res.body = req.method + " Hello World"
+		  res.status = 200
+		end
+		puts("foo")
+		server.start
+		`,
+			"GET Hello World",
+			200},
+	}
+
+	for _, tt := range tests {
+		go func() {
+			v := initTestVM()
+			v.testEval(t, tt.input, getFilename())
+		}()
+
+		time.Sleep(1 * time.Second)
```

Comment:
This is my only concern. Would running `server.start` from a thread cause execution to end in the test VM? If not, you could use a mutex.

Reply:
What do you mean? It'll make the vm inside goroutine to run the server (and hang there). This sleep is just to make sure we don't make the request before the server is started

Reply:
I metioned the wrong syncing mechanism, meant a [wait group](https://golang.org/pkg/sync/#WaitGroup). I just think we shouldn't rely on `time.sleep` for thread safety. I was curious if the following snippets would work:

```ruby
thread do
   server.start
end
```
the tests function 
```go
waitgroup.Add(1)
go func() {
    ...
    v.testEval(...) // does the vm die after this statement? or will the server still be running?
    waitgroup.Done()
}

waitgroup.Wait()
// run the tests

Reply:
@ear7h The problem is that since we're running a web server in `testEval`, so it won't actually end and call `waitgroup.Done()`

Reply:
Ah okay, I wasn't sure of the mechanics of the test vm